### PR TITLE
Logging and MLFlow enhancements & fixes

### LINKED ISSUE #1736: [BUG] Issue with Mlflow Timeseries_beta branch
if in setup log_plot set True then it is giving error in self._mlflow_log_model() as 
for plot in log_plots:
TypeError: 'bool' object is not iterable

### PR DESCRIPTION
**Pull request by ngupta23**
## Related Issue or bug
  - Fixes #1736 
  - Implements #1782
  - Implements #1708 

#### Describe the changes you've made
- Implemented `save` option for plot models
- Fixed ability to add plots to MLFlow artifacts for time_series
- Refactored unit tests for plotting (modularization)
- Added unit test for MLFlow

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Unit tests have been added and updated

## Checklist:
- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by ngupta23** [comment]
@satya-pattnaik FYI... I made some changes to the `plot_` functionality. Essentially, this will be responsible for only creating the `plotly` figure and returning it along with the data used to create it (always returns these 2 items).

The `show` functionality will be handled outside (from where `plot_` is called) since we need to interface this with streamlit as well (per @moezali1)

```python
return_data_dict = {
    "train": train,
    "test": test,
}

return fig, return_data_dict
```

File: pycaret/internal/pycaret_experiment/time_series_experiment.py [hunk 1]
```diff
@@ -2309,12 +2349,13 @@ def plot_model(
...
         plot: Optional[str] = None,
         return_data: bool = False,
+        verbose: bool = False,
         display_format: Optional[str] = None,
         data_kwargs: Optional[Dict] = None,
         fig_kwargs: Optional[Dict] = None,
         system: bool = True,
         save: Union[str, bool] = False,
-    ) -> str:
+    ) -> Tuple[Any]:
```
Comment (Yard1) [inline pycaret/internal/pycaret_experiment/time_series_experiment.py:2358]:
```suggestion
    ) -> Tuple[str, Any]:
```

File: pycaret/internal/pycaret_experiment/time_series_experiment.py [hunk 2]
```diff
@@ -2359,6 +2400,10 @@ def plot_model(
             When set to True, it returns the data for plotting.
 
 
+        verbose: bool, default = True
+            Unused for now
+
+
```
Comment (Yard1) [inline pycaret/internal/pycaret_experiment/time_series_experiment.py:2406]: One line too many

File: pycaret/time_series.py [hunk 3]
```diff
@@ -877,16 +877,17 @@ def plot_model(
     estimator: Optional[Any] = None,
     plot: Optional[str] = None,
     return_data: bool = False,
+    verbose: bool = False,
     display_format: Optional[str] = None,
     data_kwargs: Optional[Dict] = None,
     fig_kwargs: Optional[Dict] = None,
     save: Union[str, bool] = False,
-) -> str:
+) -> Tuple[Any]:
```
Comment (Yard1) [inline pycaret/time_series.py:885]:
```suggestion
) -> Tuple[str, Any]:
```

File: pycaret/time_series.py [hunk 4]
```diff
@@ -921,11 +922,15 @@ def plot_model(
...
-        
+
     return_data: bool, default = False
         When set to True, it returns the data for plotting.
 
 
+    verbose: bool, default = True
+            Unused for now
+
+
```
Comment (Yard1) [inline pycaret/time_series.py:933]: One line too many

**Review by Yard1 — APPROVED** [review]
Thanks! Left a few nits

**Review by moezali1 — APPROVED** [review]

**Pull request merged**